build-support/integ_test_android.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+if [ ! -d android ]; then
+    echo "No Android project to test" >&2
+    exit
+fi
+
+TARGET=integration_test/main_test.dart
+if [ ! -e $TARGET ]; then
+    echo "$TARGET file not found" >&2
+    exit
+fi
+
+flutter test \
+    --no-pub \
+    -d sdk \
+    $TARGET
+
+TEST_ENTRIES=`ls integration_test/separate_integration_tests/*.dart`
+for ENTRY in $TEST_ENTRIES
+do
+    echo "Run $ENTRY WITH API Sync"
+    flutter test \
+        --no-pub \
+        --dart-define ENABLE_CLOUD_SYNC=true \
+        -d sdk \
+        $ENTRY
+done
```

Comment:
request: Could you please make this script such that it will have a non-zero exit status if any test command fails? I just tried forcing the first test suite to fail and and forcing the rest to pass, and even though output says "some tests failed" from failed test suite, melos thinks it passed, I think bc exit status of this script is still 0 if last suite it runs passed.

A good way to do this is to keep a list/array of failed tests, appending to it every time the `flutter test` command fails. Then, make the exit status the length of the list. no fails = status 0. You can see an example of that in test_all_plugins.sh

Same request for ios script.